tests/ottest/fixtures/blockchain/RegtestSimple.cpp

```diff
@@ -469,11 +473,60 @@ void Regtest_fixture_simple::SendCoins(
     std::cout << sender.name_ + " send " << coins_to_send << " to "
               << receiver.name_ << " address: " << address << std::endl;
 
+    const auto& network_receiver =
+        receiver.api_->Network().Blockchain().GetChain(test_chain_);
+    auto output_size =
+        network_receiver.Wallet()
+            .GetOutputs(
+                receiver.nym_->ID(), blockchain::node::TxoState::ConfirmedNew)
+            .size();
     auto future = network.SendToAddress(
         sender.nym_id_, address, coins_to_send, memo_outgoing);
 
     // We need to confirm send transaction, so it is available after restore
     MineTransaction(sender, future.get().second, current_height);
+    WaitForOutputs(receiver, send_transactions_, output_size);
+}
+
+auto Regtest_fixture_simple::WaitForOutputs(
+    const User& receiver,
+    std::vector<Transaction>& transactions,
+    const size_t output_size) -> void
+{
+    const auto& network =
+        receiver.api_->Network().Blockchain().GetChain(test_chain_);
+
+    auto begin = std::chrono::steady_clock::now();
+    auto now = begin;
+    auto end = begin + wait_time_limit_;
+
+    for (auto& transaction : transactions) {
+        while (now < end) {
+            auto wallet_transactions = network.Wallet().GetTransactions();
+
+            if (std::find(
+                    wallet_transactions.begin(),
+                    wallet_transactions.end(),
+                    transaction->ID()) == wallet_transactions.end()) {
+                continue;
```

Comment:
Won't this result in a very tight waiting loop? Perhaps at least some milliseconds sleep before trying again?

Reply:
Fixed